packages/medusa/src/api-v2/admin/products/route.ts

```diff
@@ -1,22 +1,58 @@
+import { createProductsWorkflow } from "@medusajs/core-flows"
+import { CreateProductDTO } from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  isString,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
 import {
   AuthenticatedMedusaRequest,
   MedusaResponse,
 } from "../../../types/routing"
-
-import { CreateProductDTO } from "@medusajs/types"
-import { createProductsWorkflow } from "@medusajs/core-flows"
-import { remoteQueryObjectFromString } from "@medusajs/utils"
+import { listPriceLists } from "../price-lists/queries"
 
 export const GET = async (
   req: AuthenticatedMedusaRequest,
   res: MedusaResponse
 ) => {
-  const remoteQuery = req.scope.resolve("remoteQuery")
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const filterableFields = { ...req.filterableFields }
+  const filterByPriceListIds =
+    (filterableFields.price_list_id as string[]) || []
```

Comment:
Are we sure this will be an array? Should we instead of `Array.isArray(...) ? pricelist : [pricelist]`?

Reply:
Yes, its defined in the query validator to always be an array inside validators.ts